dev/bots/service_worker_test.dart

```diff
@@ -562,3 +568,89 @@ Future<void> runWebServiceWorkerTestWithCachingResources({
 
   print('END runWebServiceWorkerTestWithCachingResources(headless: $headless, testType: $testType)\n');
 }
+
+Future<void> runWebServiceWorkerTestWithBlockedServiceWorkers({
+  required bool headless
+}) async {
+  final Map<String, int> requestedPathCounts = <String, int>{};
+  void expectRequestCounts(Map<String, int> expectedCounts) =>
+      _expectRequestCounts(expectedCounts, requestedPathCounts);
+
+  AppServer? server;
+  Future<void> waitForAppToLoad(Map<String, int> waitForCounts) async =>
+      _waitForAppToLoad(waitForCounts, requestedPathCounts, server);
+
+  Future<void> startAppServer({
+    required String cacheControl,
+  }) async {
+    final int serverPort = await findAvailablePort();
+    final int browserDebugPort = await findAvailablePort();
+    server = await AppServer.start(
+      headless: headless,
+      cacheControl: cacheControl,
+      // TODO(yjbanov): use a better port disambiguation strategy than trying
+      //                to guess what ports other tests use.
+      appUrl: 'http://localhost:$serverPort/index.html',
```

Comment:
I know this was copied from somewhere else, but I'm not sure what the comment means. I don't see any port guessing going on. We are just finding a "random" available port and using it.

Reply:
Yes, this was copied from a few lines above. I think @yjbanov wants a better way to determine an open port than picking a random number until you find one where the OS lets you start listening!

Reply:
I have no idea what I wrote there :)